### PR TITLE
Refactor dApp Structure 06 | Finale Provider Update

### PR DESCRIPTION
**Pull request by Da-Colon**
## Description
I'll start by saying this is a large amount of file changes. but all these files existed before and only been moved. I will point out the manager changes and files in a review that will follow shortly. 

Moves all currently implemented module code to the FractalProvider. 
- Adds Governance and Treasury reducers, dispatch, actions, etc to store data to use for the app
- Puts hooks in place to load data when Safe address is validated.

## Notes

This puts the app back in a working state to where we can build the the reducers from here to store data/func to be used throughout the app.

- I also noticed that a toast would show saying that the address is invalid. I will look into this later

### CODE REVIEW

File: src/providers/fractal/hooks/useCreateGnosisDAODataCreator.ts [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+  DAO__factory,
+  DAOAccessControl__factory,
+  ERC1967Proxy__factory,
+} from '@fractal-framework/core-contracts';
+import { ERC1967Proxy__factory as GnosisERC1967Proxy__factory } from '../../../assets/typechain-types/gnosis-wrapper';
+import { TrustedAddress } from '../../../components/DaoCreator/provider/types';
+import { Interface } from 'ethers/lib/utils';
+import { getRandomBytes } from '../../../helpers';
+
+const useCreateGnosisDAODataCreator = () => {
```
Comment (Da-Colon) [inline src/providers/fractal/hooks/useCreateGnosisDAODataCreator.ts:16]: Just moving the file

File: src/providers/fractal/models/Web3NFT.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { BigNumber, ethers } from 'ethers';
+import {
+  IERC721Metadata,
+  IERC721Metadata__factory,
+} from '../../../assets/typechain-types/module-treasury';
+export class Web3NFT {
```
Comment (Da-Colon) [inline src/providers/fractal/models/Web3NFT.ts:6]: We will still need these. files for USUL

File: src/providers/fractal/reducers/gnosis.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+export const initializeGnosisState = (_initialState: IGnosis) => {
+  return _initialState;
+};
+
+export const gnosisReducer = (state: IGnosis, action: GnosisActions): IGnosis => {
+  switch (action.type) {
+    case GnosisAction.SET_SAFE:
+      return { ...state, safe: { ...action.payload }, isGnosisLoading: false };
+    case GnosisAction.SET_MODULES:
+      return { ...state, modules: action.payload, isGnosisLoading: false };
```
Comment (Da-Colon) [inline src/providers/fractal/reducers/gnosis.ts:13]: I moved the modules array to the state and added a dispatch and action for it.

File: src/providers/fractal/reducers/governance.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import { IGovernance } from './../types/governance';
+
+export const initializeGovernanceState = (_initialState: IGovernance) => {
+  return _initialState;
+};
+
+export const governanceReducer = (state: IGovernance, action: GovernanceActions): IGovernance => {
+  switch (action.type) {
+    case GovernanceAction.ADD_GOVERNANCE_DATA:
+      return { ...state, ...action.payload };
```
Comment (Da-Colon) [inline src/providers/fractal/reducers/governance.ts:12]: This action is called by `useGnosisGovernance`. think of this as the update to the Governance Injector. that handles identifying the governance and setting the state object

**Review by Da-Colon — COMMENTED** [review]
Phew. What a series of PRs. 

So where does this PR leave us.
- Tested Pure Gnosis create workflow: Success!
- Tested that DAO is properly loading in state: Success!

- There is probably still some more deleting to go but this is a good safe start

File: src/routes/DAOSubRoutes.tsx [hunk 5]
```diff
@@ -1,35 +1,38 @@
...
+        <Route
+          path={DAO_ROUTES.activties.path}
+          element={<div />}
+        />
+        <Route
+          path={DAO_ROUTES.treasury.path}
+          element={<div />}
+        />
+        <Route
+          path={DAO_ROUTES.proposals.path}
```
[the remark this answers is not in the document]
Comment (Da-Colon) [reply src/routes/DAOSubRoutes.tsx:23]: I don't have any plans in this series, After this series. the structure should be in placed that we can all start tackling these other pages and putting them back in place

File: src/providers/fractal/hooks/useSubDAODeploy.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import { useTransaction } from '../../../contexts/web3Data/transactions';
+
+import { GnosisDAO, DeployDAOSuccessCallback, GovernanceTypes } from '../types';
+
+export const useSubDAODeploy = () => {
+  const [contractCallDeploy, deploySubDAOPending] = useTransaction();
+
+  const deploySubDao = useCallback(
+    (daoData: GnosisDAO, successCallback: DeployDAOSuccessCallback) => {
+      switch (daoData.governance) {
```
[the remark this answers is not in the document]
Comment (Da-Colon) [reply src/providers/fractal/hooks/useSubDAODeploy.ts:18]: I think its safe without one since nothing would happen anyway.

**Review by herbig — APPROVED** [review]
🚀🚀🚀

**Pull request merged**